### ISSUE TITLE
.create requires .save

### ISSUE THREAD

**Issue by jrhorn424** [ga-wdi-boston/rails-summative-adv-08#12]
Note to self

**Comment by gaand** [comment]
For creating two (or more) associated models or is this about something else?  Create should save except for that situation.

**Comment by jrhorn424** [comment]
The context was creating an instance of a join model. Is that what you mean?

If so I'll update the ticket.

**Comment by GA-MEB** [comment]
@jrhorn424 if I remember correctly, the issue wasn't that `create` requires `save`, but rather that `create` returns an ActiveRecord object while `save` returns a boolean; that's why we need to set up the `create` controller action as 

``` ruby
def create
  movie = Movie.create(movie_params)
  if movie.save
    render json: movie
  else
    render json: movie.errors, status: :unprocessable_entity
  end
end
```

**Comment by gaand** [comment]
Doesn't this work?:

``` ruby
def create
  movie = Movie.create(movie_params)
  if movie.valid?
    render json: movie
  else
    render json: movie.errors, status: :unprocessable_entity
  end
end
```

**Comment by jrhorn424** [comment]
Yes, @MattB-GA-Boston that rings a bell. Also, I like @gaand's code better than what I taught.

But I'm wondering now whether we'd need to call `.save` rather than `.valid?` in this situation:

``` ruby
def create
  casting = Casting.create(casting_params))
  if casting.valid?
    render json: casting
  else
    render json: casting.errors, status: :unprocessable_entity
  end
end

private

def casting_params
  params.require(:casting).permit(:person_id, :movie_id)
end
```

**Comment by gaand** [comment]
Unless there's a functional difference I'm missing,  `save` is unnecessary.  `create` saves if `valid?`.  And `valid?` seems more semantic.

**Comment by jrhorn424** [comment]
OK, I see what you mean. But what did you mean by:

> For creating two (or more) associated models or is this about something else? Create should save except for that situation.

Did you mean this?

```
Casting.create!(
  { },
  { },
  # etc.
)
```

**Comment by gaand** [comment]
I think I may have mistyped. I think I meant the circumstance where you creating e.g. user and profile and you want to build a user, then an associated profile then call save on the user with autosave true on the relation.  For the join model situation, if you've got the ids create should suffice

**Comment by jrhorn424** [comment]
Ah, thanks! 😄
